packages/components/src/components/icon/icon.tsx

```diff
@@ -19,12 +19,12 @@ const Icon = React.forwardRef(
             onMouseLeave,
             size = 16,
             width,
-        },
-        ref
+        }: TIconProps,
+        ref?: React.ForwardedRef<SVGSVGElement | null>
```

Comment:
why did you make ref optional? :thinking:

Reply:
This is due to the failures in this PR. i.e. not all places that use the component pass in the ref prop